### ISSUE TITLE
Benchmarking

### ISSUE THREAD

**Issue by stephenstubbs** [LamaAni/PGXL-HELM#41]
I will setup a benchmarking script based on this https://github.com/gregs1104/pgbench-tools. https://emerichunter.github.io/pgbench-tools-p1/

**Comment by LamaAni** [comment]
I think we should expand on this issue once you have the basic tests in. I can do that if you post docs about the steps you took.

**Comment by stephenstubbs** [comment]
ok great

**Comment by stephenstubbs** [comment]
This is almost done. Should I create the benchmarking folder in the base path or experiemental?

**Comment by LamaAni** [comment]
That would be up to you,

If you want this to be in the base folder then I would have to adhere to a more strict way of PR approval. Sorry about that but this is the only way to keep order. 

If you like to see examples of these (though I need to further clean the my own code) you can find them in scripts. In general,
1. Code should follow the shellcheck methodology for writing. 
2. Code should have comments about what happens.
3. Code should be clean (no horz lines, etc).
4. Files should be kept small, if possible. Extended functions seperated to diffrent files and included with source. 
5. Env variables should long descriptive names. 
6. Definition files should be always read from an external source, unless a single line (yaml, txt, sql etc...)

In the future I intend to add automatic styling, but that is not the case now.

**Comment by LamaAni** [comment]
Also, @sstubbs , if you would like to have a more direct conversations, you can find me here: https://www.linkedin.com/in/zav-shotan-86a30456/ 

(I will remove this message in a few days so it dose not hang).

**Comment by stephenstubbs** [comment]
OK let me let move the tests and add the benchmarks to experimental for now. Once you happy with everything we can move them to the base at a future point.

**Comment by stephenstubbs** [comment]
I've sent you an invitation on linkedin.

**Comment by LamaAni** [comment]
Sounds good, may I make changes or reorder? If so can I write on the PR branch?

**Comment by stephenstubbs** [comment]
Yes sure. Both the testing and benchmarking are just my initial try at this. Feel free to change anything you want and let me know anything you want me to do.
